libcaf_net/caf/net/octet_stream/flow_bridge.test.cpp

```diff
@@ -0,0 +1,264 @@
+  bridge_ptr->init(ptr.get());
+  auto res = bridge_ptr->publisher();
+  mpx->start(ptr);
+  // Receive the results. We wait a bit here to subscribe after the bridge has
+  // been fully set up in order to trigger more code paths.
+  std::this_thread::sleep_for(10ms);
+  auto received = std::make_shared<std::vector<int>>();
+  auto self = flow::scoped_coordinator::make();
+  res.observe_on(self.get()).for_each([received](int item) {
+    received->push_back(item);
+  });
+  self->run();
+  // Check the results.
+  auto want = std::vector<int>();
+  want.resize(50);
+  std::iota(want.begin(), want.end(), 2);
+  check_eq(*received, want);
+  // Cleanup.
+  ping_pong_thread.join();
+}
+
+TEST("a bridge may use time-based flow operators") {
+  // Socket and thread setup.
+  auto fd_pair = net::make_stream_socket_pair();
+  require(fd_pair.has_value());
+  auto [fd1, fd2] = *fd_pair;
+  auto ping_pong_thread = std::thread{ping_pong, fd2, 5};
+  // Bridge setup.
+  auto received = std::make_shared<std::vector<int>>();
+  auto rendezvous = detail::latch{2};
+  auto bridge = net::octet_stream::flow_bridge::make(
+    16, 16,
+    [received, &rendezvous](flow::observable<std::byte> bytes) {
+      bytes.map([](std::byte item) { return std::to_integer<int>(item); })
+        .do_finally([&rendezvous] { rendezvous.count_down(); })
+        .for_each([received](int item) { received->push_back(item); });
+    },
+    [](flow::coordinator* self) {
+      return self->make_observable()
+        .interval(10ms) //
+        .take(5)
+        .map([](int64_t x) { return std::byte{static_cast<uint8_t>(x)}; });
+    });
+  auto bridge_ptr = bridge.get();
+  auto transport = net::octet_stream::transport::make(fd1, std::move(bridge));
+  transport->active_policy().connect();
+  auto ptr = net::socket_manager::make(mpx.get(), std::move(transport));
+  bridge_ptr->init(ptr.get());
+  mpx->start(ptr);
+  // Check the results.
+  rendezvous.count_down_and_wait();
+  auto want = std::vector<int>();
+  want.resize(5);
+  std::iota(want.begin(), want.end(), 1);
+  check_eq(*received, want);
+  // Cleanup.
+  ping_pong_thread.join();
+}
+
+TEST("passing never to a flow bridge omits outputs") {
+  // Socket and thread setup.
+  auto fd_pair = net::make_stream_socket_pair();
+  require(fd_pair.has_value());
+  auto [fd1, fd2] = *fd_pair;
+  auto ping_pong_thread = std::thread{iota_send, fd2, 1024};
+  // Bridge setup.
+  auto received_total = std::make_shared<size_t>();
+  auto rendezvous = detail::latch{2};
+  auto bridge = net::octet_stream::flow_bridge::make(
+    16, 16,
+    [received_total, &rendezvous](flow::observable<std::byte> bytes) {
+      bytes.map([](std::byte item) { return std::to_integer<int>(item); })
+        .do_finally([&rendezvous] { rendezvous.count_down(); })
+        .for_each([received_total](int) { *received_total += 1; });
+    },
+    [](flow::coordinator* self) {
+      return self->make_observable().never<std::byte>();
+    });
```

Comment:
I find the notation a bit confusing, specially with the phrase `... omits ...` in the test description. Could you just confirm my reasoning: The first lambda are `inputs` as in inputs from the flow to my application, and the second lambda refers to `outputs` as in my application outputs to the flow?